core/services/functions/listener.go

```diff
@@ -157,39 +160,78 @@ func NewFunctionsListener(
 	decryptor threshold.Decryptor,
 ) *FunctionsListener {
 	return &FunctionsListener{
-		oracle:          oracle,
-		oracleHexAddr:   oracle.Address().Hex(),
-		job:             job,
-		bridgeAccessor:  bridgeAccessor,
-		logBroadcaster:  logBroadcaster,
-		mbOracleEvents:  utils.NewHighCapacityMailbox[log.Broadcast](),
-		chStop:          make(chan struct{}),
-		pluginORM:       pluginORM,
-		pluginConfig:    pluginConfig,
-		s4Storage:       s4Storage,
-		logger:          lggr,
-		mailMon:         mailMon,
-		urlsMonEndpoint: urlsMonEndpoint,
-		decryptor:       decryptor,
+		client:             client,
+		contractAddressHex: contractAddressHex,
+		contractVersion:    pluginConfig.ContractVersion,
+		job:                job,
+		bridgeAccessor:     bridgeAccessor,
+		logBroadcaster:     logBroadcaster,
+		mbOracleEvents:     utils.NewHighCapacityMailbox[log.Broadcast](),
+		chStop:             make(chan struct{}),
+		pluginORM:          pluginORM,
+		pluginConfig:       pluginConfig,
+		s4Storage:          s4Storage,
+		logger:             lggr,
+		mailMon:            mailMon,
+		urlsMonEndpoint:    urlsMonEndpoint,
+		decryptor:          decryptor,
 	}
 }
 
+func (l *FunctionsListener) ContractVersion() uint32 {
```

Comment:
We don't need this. Tests should know in advance which version is used because they are building the config too. You can add a version param to NewFunctionsListenerUniverse().